File: atsa/src/main/java/jp/kaiz/atsassistmod/registry/ATSAModBlocks.java
package jp.kaiz.atsassistmod.registry;

import jp.kaiz.atsassistmod.ATSAssistMod;
import jp.kaiz.atsassistmod.block.GroundUnitBlock;
import jp.kaiz.atsassistmod.block.IftttBlock;
import jp.kaiz.atsassistmod.block.StationAnnounceBlock;
import net.minecraft.world.level.block.state.BlockBehaviour;
import net.neoforged.neoforge.registries.DeferredBlock;
import net.neoforged.neoforge.registries.DeferredRegister;

public final class ATSAModBlocks {
    public static final DeferredRegister.Blocks BLOCKS = DeferredRegister.createBlocks(ATSAssistMod.MODID);

    public static final DeferredBlock<GroundUnitBlock> GROUND_UNIT =
            BLOCKS.registerBlock("groundunit", GroundUnitBlock::new,
                    () -> BlockBehaviour.Properties.of().strength(1.5F, 6.0F).requiresCorrectToolForDrops());
    public static final DeferredBlock<IftttBlock> IFTTT =
            BLOCKS.registerBlock("ifttt", IftttBlock::new,
                    () -> BlockBehaviour.Properties.of().strength(1.5F, 6.0F).requiresCorrectToolForDrops());
    public static final DeferredBlock<StationAnnounceBlock> STATION_ANNOUNCE =
            BLOCKS.registerBlock("station_announce", StationAnnounceBlock::new,
                    () -> BlockBehaviour.Properties.of().strength(1.5F, 6.0F).requiresCorrectToolForDrops());

    private ATSAModBlocks() {}

    public static void register(net.neoforged.bus.api.IEventBus bus) {
        BLOCKS.register(bus);
    }
}

File: atsa/src/main/kotlin/jp/kaiz/atsassistmod/registry/ATSAModBlocks.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package jp.kaiz.atsassistmod.registry

import jp.kaiz.atsassistmod.ATSAssistMod
import jp.kaiz.atsassistmod.block.GroundUnitBlock
import jp.kaiz.atsassistmod.block.IftttBlock
import jp.kaiz.atsassistmod.block.StationAnnounceBlock
import net.minecraft.world.level.block.state.BlockBehaviour
import net.neoforged.bus.api.IEventBus
import net.neoforged.neoforge.registries.DeferredBlock
import net.neoforged.neoforge.registries.DeferredRegister
import java.util.function.Function
import java.util.function.Supplier

object ATSAModBlocks {
    @JvmField
    val BLOCKS: DeferredRegister.Blocks = DeferredRegister.createBlocks(ATSAssistMod.MODID)

    @JvmField
    val GROUND_UNIT: DeferredBlock<GroundUnitBlock> = BLOCKS.registerBlock(
        "groundunit",
        Function(::GroundUnitBlock),
        Supplier {
        BlockBehaviour.Properties.of().strength(1.5f, 6.0f).requiresCorrectToolForDrops()
        },
    )

    @JvmField
    val IFTTT: DeferredBlock<IftttBlock> = BLOCKS.registerBlock(
        "ifttt",
        Function(::IftttBlock),
        Supplier {
            BlockBehaviour.Properties.of().strength(1.5f, 6.0f).requiresCorrectToolForDrops()
        },
    )

    @JvmField
    val STATION_ANNOUNCE: DeferredBlock<StationAnnounceBlock> = BLOCKS.registerBlock(
        "station_announce",
        Function(::StationAnnounceBlock),
        Supplier {
            BlockBehaviour.Properties.of().strength(1.5f, 6.0f).requiresCorrectToolForDrops()
        },
    )

    @JvmStatic
    fun register(bus: IEventBus) {
        BLOCKS.register(bus)
    }
}
